### ISSUE TITLE
Add support for disabling dashboard through `appsettings.json`

### ISSUE THREAD

**Issue by abjerner** [skybrud/Skybrud.Umbraco.Redirects#162]

**Comment by richard-arkom** [comment]
Hi,

Is this something that is available now, or will it be added to a future version?

Thanks

**Comment by abjerner** [comment]
@abjerner not at the moment, no. Are you looking for controlling which user groups has access to the dashboard, or do you want to disable the dashboard for all users?

I played around with this at some point, as I wanted to support also being able to set up the permissions via the app settings. I recall I had some problems finding the correct format, and I haven't worked on this for some time.

But being able to disable the dashboard entirely should be fairly straight forward. I will likely push a new release later today, and I can see if I can include an option for this 😉

**Comment by richard-arkom** [comment]
Hi,

Yes, looking to control which user groups have access to the dashboard, ideally only show the 'Redirects' dashboard for administrators. But disabling entirely would be a good backup option that we could enable once we'd added our initial redirects.

Cheers

**Comment by abjerner** [comment]
For now you can control this from C#:

https://packages.skybrud.dk/skybrud.umbraco.redirects/docs/v4/configuration/dashboard/

https://packages.skybrud.dk/skybrud.umbraco.redirects/docs/v4/configuration/#backoffice-helper

**Comment by abjerner** [comment]
And this will now be possible in the next release:

```json
{
  "Skybrud": {
    "Redirects": {
      "Dashboard": {
        "Enabled": false
      }
    }
  }
}
```

I have another issue in the package that I need to look into, but hopefully I'll have `v4.0.9` out later today.

Ideally it should also be possible to configure the permissions from appsettings.json, but I'll need some more time to look into that, so this will have to wait for another future release.

**Comment by richard-arkom** [comment]
Ok. That's great.

Would it also be possible to disable the [Redirects] button which shows between [Content] and [i] on the top right, when editing a content node.

Cheers

**Comment by abjerner** [comment]
The content app as in this screenshot?

![image](https://github.com/skybrud/Skybrud.Umbraco.Redirects/assets/3634580/6c599ea8-c6a8-4e75-8da8-bb79ef245c5e)

If so, it can already be disabled via `appsettings.json`:

```json
{
  "Skybrud": {
    "Redirects": {
      "ContentApp": {
        "Enabled": false
      }
    }
  }
}
```

Or you can opt-in or opt-out of specific content or media types. Eg. here I've opted out for media in general, but opted in for images specifically (since they would otherwise be hit be the wildcard rule):

```json
{
  "Skybrud": {
    "Redirects": {
      "ContentApp": {
        "Enabled": true,
        "Show": [
          "-content/site",
          "-content/subSite",
          "-media/*",
          "-media/vimeoVideo",
          "+media/Image"
        ]
      }
    }
  }
}
```

Unfortunately this isn't documented at the moment, although it should be 😊

**Comment by richard-arkom** [comment]
Awesome 👍

**Comment by abjerner** [comment]
Closing the issue as this was implemented in [**`v4.0.9`**](https://github.com/skybrud/Skybrud.Umbraco.Redirects/releases/tag/v4.0.9).

Commit: https://github.com/skybrud/Skybrud.Umbraco.Redirects/commit/a4b2e3c10561857de43063be6551750f9f02d677
